string color
int32 x1
int32 y1
int32 x2
int32 y2
---
float32[2] center
float32[2] size
float32 angle
